aa_stripe/models.py

```diff
@@ -133,6 +133,16 @@ def __str__(self):
 
     def save(self, *args, **kwargs):
         stripe.api_key = settings.STRIPE_API_KEY
+        if self.is_deleted:
```

Comment:
Please check previous value of is_deleted and update it only when it was changed to true. 
Check how https://github.com/ArabellaTech/getreferd-api/blob/develop/getreferd/accounts/models.py#L40 this works here.